### PR TITLE
Some docs revisions

### PR DESCRIPTION
**Pull request by ttmc**
Some mostly-minor editing:

* Updated the docstring of the `__init__()` method of the `Bigchain` class, to bring it in line with how configuration settings are actually determined today
* Moved the docs about JSON Serialization before the docs about Cryptography, because conceptually one must serialize before calculating hashes or signatures
* Lots of copy-editing of the Python Server API Examples

### CODE REVIEW

**Comment by vrde** [comment]
@ttmc I fixed a typo. For me this PR is ready to get merged. Great job in fixing the docs!

**Comment by ttmc** [comment]
@diminator I would like you to review the changes I made to the Python Server API Examples (in the documentation) before we merge this PR.

File: docs/source/python-server-api-examples.md [hunk 1]
```diff
@@ -750,9 +749,13 @@ threshold_tx_transfer
 
 ### Hash-locked Conditions
 
-By creating a hash of a difficult-to-guess 256-bit random or pseudo-random integer it is possible to create a condition which the creator can trivially fulfill by publishing the random value. However, for anyone else, the condition is cryptographically hard to fulfill, because they would have to find a preimage for the given condition hash.
+A hash-lock condition on an asset is like a password condition: anyone with the secret preimage (like a password) can fulfill the hash-lock condition and transfer the asset to themselves.
+
+Under the hood, fulfilling a hash-lock condition amounts to finding a number (a "preimage") which, when hashed, results in a given value. It's easy to verify that a given preimage hashes to the given value, but it's computationally difficult to _find_ a number which hashes to the given value. The only practical way to get a valid preimage is to get it from the original creator (possibly via intermediaries).
```
Comment (diminator) [inline docs/source/python-server-api-examples.md]: > to finding a number

any arbitrary string actually

**Comment by diminator** [comment]
really good, shall I provide the sources for the images? Or do we care that the hash in the images is different as in the text?

File: docs/source/python-server-api-examples.md (hunk 1 above)
Comment (ttmc) [inline docs/source/python-server-api-examples.md]: I suspected that any arbitrary string could be the preimage, but the original text used the words "integer" and "value". I will push a change to this.

**Comment by ttmc** [comment]
What format are the sources for the images?

In any case, the images should be consistent with the text, otherwise someone might get confused (or think we're sloppy).

**Comment by ttmc** [comment]
I edited the two images in the Python Server API Examples section of the docs.

I'll merge this PR if all the checks pass.

**Pull request merged**